### DISCUSSION TITLE
I  am trying to build c++ std modules with latest clang without success

### DISCUSSION THREAD

**Discussion by gibberishCode** [xmake-io/xmake#4084] [Q&A]
```
add_rules("mode.debug", "mode.release")
set_languages("c++latest")

target("mod")
    set_kind("static")
    add_files("src/*.cpp", "src/*.mpp")
    set_policy("build.c++.clang.stdmodules", true)

target("stdmodules")
    set_kind("binary")
    add_files("test/*.cpp")
    add_deps("mod")
    set_policy("build.c++.clang.stdmodules", true)
```
For some reason build works only once

$ xmake 
[  0%]: generating.module.deps src/main.cpp
[ 50%]: compiling.release src/main.cpp
[ 66%]: linking.release stdmodules_cpp_only
[100%]: build ok, spent 0.341s
$ xmake 
[ 50%]: compiling.release src/main.cpp
[ 66%]: linking.release stdmodules_cpp_only
error: /usr/bin/ld: cannot find -lstdc++: No such file or directory
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by waruqi** [comment]
try 

```console
xmake f -c
xmake -rvD
```

and let me see the log.


@Arthapz Any idea?

**Reply by gibberishCode** [reply]
Sorry messed with my setup and now fail to reproduce this problem. Although I still failed to compile std modules "out of the box" using xmake I will listen to Arthapz and try Clang modules. Thank you both for your work

**Comment by Arthapz** [comment]
If you really want to use the std modules i would suggest to use the official clang c++23 std modules and not the clang module emulation that we implemented in XMake (that are not C++20 modules) 

https://github.com/llvm/llvm-project/tree/main/libcxx/modules

i tried it with XMake and it work well, we will add the support on XMake when llvm ship these modules (actually they only present in source tree, not in final llvm installation)

and if you are using clang on windows i'm currently checking if we can use the MSVC std module with clang
